src/scratchOrgSettingsGenerator.ts

```diff
@@ -210,62 +172,168 @@ export default class SettingsGenerator {
+              {
+                ...allRecordTypes.reduce((acc, allRecordType) => {
+                  return {
+                    ...acc,
+                    ...{ members: [...(acc.members ?? []), allRecordType] },
+                  };
+                }, Object.create(null)),
+                name: 'RecordType',
+              },
+            ],
+          },
+        };
+      }
 
-  private createObjectFileContent(json: Record<string, unknown>) {
-    const output: ObjectSetting = {};
-    if (json.sharingModel) {
-      output.sharingModel = upperFirst(json.sharingModel as string);
+      if (allbusinessProcesses.length > 0) {
+        output = {
+          ...output,
+          ...{
+            types: [
+              output.types,
+              {
+                ...allbusinessProcesses.reduce((acc, allRecordType) => {
+                  return {
+                    ...acc,
+                    ...{ members: [...(acc.members ?? []), allRecordType] },
+                  };
+                }, Object.create(null)),
+                name: 'BusinessProcess',
+              },
+            ],
+          },
+        };
+      }
     }
-    return output;
+    output = { ...output, ...{ version: apiVersion } };
+    const xml = js2xmlparser.parse('Package', output);
+    this.writer.addToZip(xml, packageFilePath);
   }
 
-  private createRecordTypeFileContent(
-    objectName: string,
-    setting: ObjectSetting
-  ): { fullName: Optional<string>; label: Optional<string>; active: boolean; businessProcess?: Optional<string> } {
-    const defaultRecordType = upperFirst(setting.defaultRecordType);
-    const output = {
-      fullName: defaultRecordType,
-      label: defaultRecordType,
-      active: true,
-    };
-    // all the edge cases
-    if (['Case', 'Lead', 'Opportunity', 'Solution'].includes(upperFirst(objectName))) {
-      return { ...output, businessProcess: `${defaultRecordType}Process` };
+  private createObjectFileContent(
+    name: string,
+    json: Record<string, unknown>,
+    allRecordTypes: string[],
+    allbusinessProcesses: string[]
+  ) {
+    let output = {
+      '@': {
+        xmlns: 'http://soap.sforce.com/2006/04/metadata',
+      },
+    } as JsonMap;
+    const sharingModel = json['sharingModel'];
+    if (sharingModel) {
+      output = {
+        ...output,
+        sharingModel: upperFirst(sharingModel as string),
+      };
     }
-    return output;
-  }
 
-  private createBusinessProcessFileContent(
-    objectName: string,
-    businessProcessName: string
-  ): BusinessProcessFileContent {
-    const objectToBusinessProcessPicklist: ObjectToBusinessProcessPicklist = {
-      Opportunity: { fullName: 'Prospecting' },
-      Case: { fullName: 'New', default: true },
-      Lead: { fullName: 'New - Not Contacted', default: true },
-      Solution: { fullName: 'Draft', default: true },
-    };
+    const defaultRecordType = json['defaultRecordType'];
+    if (defaultRecordType) {
```

Comment:
if you check for type instead of truthiness, you can omit all the `defaultRecordType as string` (same with sharing model above)
```suggestion
    if (typeof defaultRecordType === 'string') {
```